### ISSUE TITLE
Cannot run without --bypass-dns-resolution=true

### ISSUE THREAD

**Issue by hznor** [postfinance/kubelet-csr-approver#198]
I'm experiencing troubles with kubelet-csr-approver without passing the option --bypass-dns-resolution=true

"Denying kubelet-serving CSR. DNS checks failed. Reason:The SAN DNS Name could not be resolved, denying the CSR"

$ dig A k8snode03u @10.96.0.10
$ dig A k8snode03u.cluster.local @10.96.0.10

Doesn't resolve to the node IP address.

Am I doing something wrong on my cluster?
Is having core-dns resolving the node's hostname a usually expected behavior?
How do you enable such feature on your cluster?

**Comment by clementnuss** [comment]
This doesn't have anything to do with your cluster config.
The node name is only resolvable **if** you have an external DNS provider (be it a DNS server, your router in some cases that are built-in with DNS resolution features, your cloud provider), or if you use the "bonjour" protocol (avahi-daemon on Linux), which permits to locally resolve node name through appending the `.local` suffix

resolving a node IP with the `cluster.local` suffix will not work, as there is no (to my knowledge) mechanism in Kubernetes and [in its CoreDNS integration](https://coredns.io/plugins/kubernetes/) to make the node IP available through a DNS query.

in summary, CoreDNS will resolve your node's IP if the upstream DNS servers can resolve your node IP. Alternatively, you may find one of CoreDNS's plugins useful in helping you achieve DNS resolution of your nodes: https://coredns.io/plugins/

I hope this answers your question.

**Comment by hznor** [comment]
Ok.

This does not seem to be a usual DNS feature on Kubernetes clusters in production.

Do you agree that this check can be safely disabled as long as a decent provider regex is configured?

**Comment by clementnuss** [comment]
yes, fully agree. the most important part is that the regex is properly defined, to prevent a node from requesting a `oauth.company.org` type of DNS name 😅
